compiler-core/src/javascript/expression.rs

```diff
@@ -120,6 +117,11 @@ pub(crate) struct Generator<'module, 'ast> {
     /// ```
     ///
     statement_level: Vec<Document<'ast>>,
+
+    /// This will be true if we've generated a statement that is guaranteed to
+    /// throw. This means we can stop code generation for all following statements
+    /// in the same block!
+    pub statement_always_throws: bool,
```

Comment:
I think this should be set to true for `panic` and `todo`. I can't see that having been done, has it?